src/Lecture1.hs

```diff
@@ -123,4 +156,37 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+
+lowerAndGreater :: Int -> [Int] -> String
+lowerAndGreater n list = 
+    let 
+        cnt = getGreaterLowerCnt n list
+    in show n ++ " is greater than " ++ show (head cnt) ++ " elements and lower than " ++ show (last cnt) ++ " elements"   
+
+getGreaterLowerCnt :: Int -> [Int] -> [Int]    
+getGreaterLowerCnt n list
+    | null list  = [0, 0] 
+    | headOrDefault list < n =
+        let 
+            cnt = [1, 0]
+        in  sumCnt cnt (getGreaterLowerCnt n (tail list))
+    | headOrDefault list > n =
+        let 
+            cnt = [0, 1]
+        in  sumCnt cnt (getGreaterLowerCnt n (tail list))
+    | otherwise = 
+        let 
+            cnt = [0, 0]
+        in  sumCnt cnt (getGreaterLowerCnt n (tail list))     
+
+sumCnt :: [Int] -> [Int] -> [Int]
+sumCnt l1 l2 = 
+    let 
+        h = head l1 + head l2
```

Comment:
the recommendation about pattern matching is about all cases where `head` or other partial functions are used :)

Reply:
I fixed it with pattern matching. Looks more safe now) Thnx